File: matching-service/src/main/java/com/joblens/matching/model/Job.java
package com.joblens.matching.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * Read/write view of the "jobs" table shared with job-aggregator-service.
 * Matching-service writes jobLevel/workMode/requiredYears back here after scoring.
 */
@Entity
@Table(name = "jobs")
@Getter
@NoArgsConstructor
public class Job {

    @Id
    private Long id;

    private String title;
    private String company;
    private String location;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(columnDefinition = "TEXT")
    private String applyUrl;

    private String source;
    private Double salaryMin;
    private Double salaryMax;
    private LocalDateTime postedDate;

    @Setter private String jobLevel;
    @Setter private String workMode;
    @Setter private Integer requiredYears;
}

File: frontend/src/types/job.ts
export interface Job {
  id: number
  externalId: string
  title: string
  company: string | null
  location: string | null
  description: string | null
  source: string
  applyUrl: string | null
  salaryMin: number | null
  salaryMax: number | null
  postedDate: string | null
  fetchedAt: string
  lastSeenAt: string | null
  jobLevel: string | null
  workMode: string | null
  requiredYears: number | null
}

export interface PageResponse<T> {
  content: T[]
  totalElements: number
  totalPages: number
  number: number
  size: number
  first: boolean
  last: boolean
}
